modules/rostests/apitests/win32nt/CMakeLists.txt

```diff
@@ -64,23 +64,44 @@ list(APPEND SOURCE
     #osver.c
     win32nt.h)
 
-add_executable(win32knt_apitest
-    ${SOURCE}
-    testlist.c
-    w32knapi.rc)
+add_library(win32knt_static STATIC ${SOURCE} w32knapi.rc)
+add_dependencies(win32knt_static xdk)
+add_pch(win32knt_static win32nt.h SOURCE)
+
+add_executable(win32knt_apitest testlist.c)
+add_executable(win32knt_xpsp2_apitest testlist.c)
+add_executable(win32knt_2k3sp2_apitest testlist.c)
+
+target_link_libraries(win32knt_apitest ${PSEH_LIB} win32knt_static gditools)
+target_link_libraries(win32knt_xpsp2_apitest ${PSEH_LIB} win32knt_static gditools)
+target_link_libraries(win32knt_2k3sp2_apitest ${PSEH_LIB} win32knt_static gditools)
 
-target_link_libraries(win32knt_apitest ${PSEH_LIB} gditools)
 set_module_type(win32knt_apitest win32cui)
-add_importlibs(win32knt_apitest
-    win32u # win32u_2ksp4 win32u_xpsp2 win32u_2k3sp2 win32u_vista
-    gdi32
-    user32
-    shell32
-    advapi32
-    msvcrt
-    kernel32
-    ntdll)
+set_module_type(win32knt_xpsp2_apitest win32cui)
+set_module_type(win32knt_2k3sp2_apitest win32cui)
+
+set(WIN32KNT_IMPORTLIBS gdi32 user32 shell32 advapi32 msvcrt kernel32 ntdll)
+
+add_importlibs(win32knt_apitest win32u ${WIN32KNT_IMPORTLIBS})
+add_importlibs(win32knt_xpsp2_apitest win32u_xpsp2 ${WIN32KNT_IMPORTLIBS})
+add_importlibs(win32knt_2k3sp2_apitest win32u_2k3sp2 ${WIN32KNT_IMPORTLIBS})
+
+if(0)  # Specify 1 if you want 2ksp4 version
```

Comment:
Suggestion for "consistency": Use an 'if' block for each version. Order them from 2k to vista.
It could be `if(1)` for default/knt and `if(0)` for all the others. Then, you can use `if(1)` locally for XP if you want it.

Reply:
Add a reminder that 2k and Vista need to be enabled in 'modules/rostests/apitests/win32u/CMakeLists.txt' too.
For current consistency, I would agree xp/2k3/ros could each be `if(1)`.

Reply:
OK.